lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -143,7 +143,7 @@ def split_authorization(authorization)
       end
 
       def generate_purchase_number_stamp
-        Time.now.to_f.to_s.delete('.')[1..10] + rand(99).to_s
+        rand(('9' * 12).to_i).to_s.center(12, rand(9).to_s)
```

Comment:
What does `center(12, rand(9).to_s)` do here?

Reply:
So the first part `rand(('9' * 12).to_i).to_s` doesn't always returns a 12-digit string so center checks if the length is 12 if not it returns a new string with added random numbers